Source/Fuse.Models/Tests/Model.Test.uno

```diff
@@ -0,0 +1,585 @@
+				root.StepFrameJS();
+				Assert.AreEqual("1,2,3", e.oc.JoinValues());
+				Assert.AreEqual("3,4,5", e.poc.JoinValues());
+				Assert.AreEqual(4, oc.Count);
+				Assert.AreEqual(4, poc.Count);
+				e.callEmpty.Perform();
+				e.callEmptyPromise.Perform();
+				root.StepFrameJS();
+				Assert.AreEqual("", e.oc.JoinValues());
+				Assert.AreEqual("", e.poc.JoinValues());
+				Assert.AreEqual(1, oc.Count);
+				Assert.AreEqual(1, poc.Count);
+			}
+		}
+
+		[Test]
+		public void MultiCounter()
+		{
+			var e = new UX.Model.MultiCounter();
+			using (var root = TestRootPanel.CreateWithChild(e))
+			{
+				root.StepFrameJS();
+				var oc = e.counterCollector.Children;
+				Assert.AreEqual(2, oc.Count);
+
+				var counter1 = oc[1] as UX.Model.Counter;
+				Assert.AreNotEqual(null, counter1);
+
+				Assert.AreEqual("0", counter1.counter.Value);
+
+				counter1.callIncrement.Perform();
+				root.StepFrameJS();
+				Assert.AreEqual("1", counter1.counter.Value);
+
+				e.callAddCounter.Perform();
+				root.StepFrameJS();
+				Assert.AreEqual(3, oc.Count);
+
+				var counter2 = oc[2] as UX.Model.Counter;
+				Assert.AreNotEqual(null, counter2);
+
+				Assert.AreEqual("0", counter2.counter.Value);
+
+				counter2.callIncrement.Perform();
+				root.StepFrameJS();
+				Assert.AreEqual("1", counter2.counter.Value);
+
+				counter2.callIncrement.Perform();
+				root.StepFrameJS();
+				Assert.AreEqual("2", counter2.counter.Value);
+
+			}
+		}
+
+		[Test]
+		public void MutatePages()
+		{
+			var e = new UX.Model.MutatePages();
+			using(var root = TestRootPanel.CreateWithChild(e))
+			{
+				root.StepFrameJS();
+				var children = ChildrenOfType<Visual>(e.navigator);
+				Assert.AreEqual(1, children.Count);
+				AssertOfType<UX.Model.MainPage>(children[0]);
+
+				e.pushPage.Perform();
+				root.StepFrameJS();
+
+				var childrenAfterPush = ChildrenOfType<Visual>(e.navigator);
+				Assert.AreEqual(2, childrenAfterPush.Count);
+				AssertOfType<UX.Model.MainPage>(childrenAfterPush[0]);
+				AssertOfType<UX.Model.DetailPage>(childrenAfterPush[1]);
+
+				e.popPage.Perform();
+				root.StepFrameJS();
+
+				AssertOfType<UX.Model.MainPage>(e.navigator.Active);
+			}
+		}
+
+		static List<T> ChildrenOfType<T>(Visual n) where T : Node
+		{
+			var l = new List<T>();
+			foreach (var child in n.Children)
+			{
+				var m = child as T;
+				if (m != null)
+					l.Add(m);
+			}
+			return l;
+		}
+
+		static void AssertOfType<T>(
+			object obj,
+			[CallerFilePath] string filePath = "",
+			[CallerLineNumber] int lineNumber = 0,
+			[CallerMemberName] string memberName = "")
+		{
+			if(!(obj is T))
+			{
+				var expected = "object of type '" + typeof(T).FullName + "'";
+				var actual = (obj == null) ? "null" : "object of type '" + obj.GetType().FullName + "'";
+
+				// Temporary: `Assert.ReportFailure` is private
+				var message = "Expected " + expected + ", got " + actual;
+				Assert.Fail(message, filePath, lineNumber, memberName);
+			}
+		}
```

Comment:
cleanup-notice: I *think* this one has made it to uno and up by now, so it can probably be removed.

Reply:
No, that doesn't seem to be true.

Reply:
Sounds like you're talking about different methods (OfType vs. ReportFailure). OfType has now landed on uno master, will switch to that